xtask/src/runchecks.rs

```diff
@@ -230,7 +230,10 @@ fn burn_core_std() {
     cargo_test(["-p", "burn-core", "--features", "test-tch"].into());
 
     // Run cargo test --features test-wgpu
-    cargo_test(["-p", "burn-core", "--features", "test-wgpu"].into());
+    // Disabled for macOS in CI due to unavailable Metal device
+    if !(std::env::var("CI_RUN").is_ok() && cfg!(target_os = "macos")) {
```

Comment:
It does work on macOS, but it doesn't work on CI agents due to Metal not being there (AFAIU).

Reply:
In my opinion you can just exclude that test (or set of tests) from all the other tests. Then you can run the excluded tests singularly when the os is different from MacOS

Reply:
I think I've found a better way to disable it thanks to your tip. Ultimately I was trying to not rework the whole testing flow and rather adapt to what we already use (xtask). IMO either we use xtask and the github workflow stays to a "minimum" or we try to move towards letting github actions carry the weight of dynamism. But doing one where part of it is here and part of it is there doesn't sound like a good outcome to me.

One benefit that relying on xtask has, is that technically we'd be free to move to any other CI service with relatively low friction.

Reply:
In case you'd like to know the error we get into:

```bash
---- tests::slice::tests::should_support_full_sliceing_1d stdout ----
thread 'tests::slice::tests::should_support_full_sliceing_1d' panicked at burn-wgpu/src/compute/base.rs:224:17:
No adapter found for graphics API AutoGraphicsApi
```

An example failed action can be found here: https://github.com/Tracel-AI/burn/actions/runs/7023591879/job/19110454885

Reply:
I realize now what you meant @Luni-4, sorry I think I misunderstood before. I think the solution I've settled on now is a good one. As in general things do work on macOS it just doesn't work on the CI. An env flag seems better suited for that.

Reply:
@dcvz 

I agree with your new solution. The last thing I would suggest for this PR is to disable `no-std` tests on `MacOS` adding these lines:

```yaml
 exclude: 
    # Do not run no-std tests on MacOS 
    - os: macos-13
      test: 'no-std'
```

Reply:
I do not know which could be the right approach here. 

@nathanielsimard this question is for you :)

Reply:
Like @Luni-4 mentionned we only need to run `no-std` tests once. I think I would still use a linux server to run them, even if they take a bit more time than on macos.

Reply:
Awesome. I’ll fix that up then :)